components/salesmate/app/salesmate.app.ts

```diff
@@ -1,13 +1,457 @@
+    }) {
+      const config = {
+        url: `${this._apiUrl({
+          path,
+          action,
+          version,
+        })}`,
+        headers: this._getHeaders(),
+        ...opts,
+      };
+
+      return axios($, clearObj(config));
+    },
+    addContact(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: "contact",
+      });
+    },
+    addDeal(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: "deal",
+      });
+    },
+    listUsers({
+      params, ...args
+    }) {
+      return this._makeRequest({
+        ...args,
+        method: "GET",
+        path: "core",
+        action: "/users",
+        params: {
+          ...params,
+          status: "active",
+        },
+      });
+    },
+    listCompanies(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: "company",
+        action: "/search",
+        data: queries.listCompanies,
+      });
+    },
+    listContacts(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: "contact",
+        action: "/search",
+        data: queries.listContacts,
+      });
+    },
+    listCurrency(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "GET",
+        path: "lookups/active/currency",
+        version: "v3",
+      });
+    },
+    listDeals(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: "deal",
+        action: "/search",
+        data: queries.listDeals,
+      });
+    },
+    listPipelines(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "GET",
+        path: "apps/dealPipeline",
+        version: "v3",
+      });
+    },
+    async listStages({
+      pipeline, ...args
+    }) {
+      return this._makeRequest({
+        ...args,
+        method: "POST",
+        path: `board-view/${pipeline}/stage-wise-summary`,
+        data: queries.listStages(pipeline),
+        version: "v1deal",
+      });
+    },
+    listTags(args = {}) {
+      return this._makeRequest({
+        ...args,
+        method: "GET",
+        path: "tags",
+        version: "v1",
+      });
+    },
+    async listVisibleFields({
+      $, module, field,
+    }) {
+      const { Data } = await this._makeRequest({
+        $,
+        method: "GET",
+        path: "fields/getAllVisibleFields",
+        version: "v3",
+      });
+      const items = Data[module];
+      return items.filter((item) => (item.fieldName === field))[0];
+    },
+    updateContact({
+      $, contactId, ...data
+    }) {
+      return this._makeRequest({
+        $,
+        method: "PUT",
+        path: "contact",
+        action: `/${contactId}`,
+        data,
+      });
+    },
+    async *paginate({
+      fn, params = {
+        rows: 0,
+        from: 0,
```

Comment:
This initialization doesn't seem to be necessary

Reply:
it's necessary for the TS validation

Reply:
This is also valid:
```js
async *paginate({
  fn, params, maxResults,
}) {
```